tests/integration/conftest.py

```diff
@@ -0,0 +1,131 @@
+    profile_class_resource = create_global_resource(
+        group="kubeflow.org", version="v1", kind="Profile", plural="profiles"
+    )
+
+    # Apply Profile first
+    apply_manifests(lightkube_client, PROFILE_FILE_PATH)
+
+    yield
+
+    # Remove namespace
+    read_yaml = Path(PROFILE_FILE_PATH).read_text()
+    yaml_loaded = codecs.load_all_yaml(read_yaml)
+    for obj in yaml_loaded:
+        try:
+            lightkube_client.delete(
+                res=type(obj),
+                name=obj.metadata.name,
+                namespace=obj.metadata.namespace,
+            )
+        except lightkube.core.exceptions.ApiError as api_error:
+            raise api_error
+
+
+@pytest.fixture(scope="session")
+def kfp_client(apply_profile, forward_kfp_ui) -> kfp.Client:
+    """Returns a KFP Client that can talk to the KFP API Server."""
+    # Instantiate the KFP Client
+    client = kfp.Client(host=KUBEFLOW_LOCAL_HOST, namespace=KUBEFLOW_PROFILE_NAMESPACE)
+    client.runs.api_client.default_headers.update({"kubeflow-userid": KUBEFLOW_USER_NAME})
+    return client
+
+
+@pytest.fixture(scope="session")
+def lightkube_client() -> lightkube.Client:
+    """Returns a lightkube Client that can talk to the K8s API."""
+    client = lightkube.Client(field_manager="kfp-operators")
+    return client
+
+
+@pytest.fixture(scope="function")
+def upload_and_clean_pipeline(kfp_client: kfp.Client):
+    """Upload an arbitrary pipeline and remove after test case execution."""
+    pipeline_upload_response = kfp_client.pipeline_uploads.upload_pipeline(
+        uploadfile=SAMPLE_PIPELINE, name=SAMPLE_PIPELINE_NAME
+    )
+
+    yield pipeline_upload_response
+
+    kfp_client.delete_pipeline(pipeline_id=pipeline_upload_response.id)
+
+
+@pytest.fixture(scope="function")
+def create_and_clean_experiment(kfp_client: kfp.Client):
+    """Create an experiment and remove after test case execution."""
+    experiment_response = kfp_client.create_experiment(
+        name="test-experiment", namespace=KUBEFLOW_PROFILE_NAMESPACE
+    )
+
+    yield experiment_response
+
+    kfp_client.delete_experiment(experiment_id=experiment_response.id)
+
+
+def pytest_addoption(parser: Parser):
+    parser.addoption(
+        "--bundle",
+        default="./tests/integration/bundles/kfp_latest_edge.yaml.j2",
+        help="Path to bundle file to use as the template for tests.  This must include all charms"
+        "built by this bundle, where the locally built charms will replace those specified. "
+        "This is useful for testing this bundle against different external dependencies. "
+        "An example file is in ./tests/integration/data/kfp_latest_edge.yaml",
+    )
+    parser.addoption(
+        "--build",
+        default=True,
+        help="Whether the charms in this repository should be built locally and used"
+        "to render the bundle definition template."
+        "If set to False, the integration tests will be run against charms in Charmhub.",
+    )
```

Comment:
This cannot be turned off:
* `tox -e bundle-integration`: results in `build=True`
* `tox -e bundle-integration -- --build`: results in `build=True`
* `tox -e bundle-integration -- --build=False`: results in `build="False"` (Note, it is a **string** of False, not the boolean)
  * This will evaluate as True [here](https://github.com/canonical/kfp-operators/pull/238/files/f602d6a13bfb39e2199c153c78d98584b885a8d1#diff-2539489e7622d481095b67a9cf5cd2e3366be6816e2ad192c9996de42c1d7487R22)

I think there is a different argument type here that you could use that lets someone do `--build` or `--no-build` to control the boolean.

Reply:
Updated to use `--no-build`, thanks for the suggestion.